Added support for black first move

trentzz opened the pull request.

donren-leung commented on checkers_bot_tournament/game.py:
```diff
@@ -93,6 +110,14 @@ def import_pdn(self, filename: str) -> None:
             self.move_number += 1
             self.swap_turn()
 
+    def move_piece(self, move: Move) -> Tuple[bool, bool]:
```
There's stuff common between the PDN making piece moves and the run loop with the bots that you could combine. For example I think if `self.verbose` mode is on you also wanna print out the move and board display. You might wanna also detect for the game ending during the PDN (assert this doesn't happen cos why would you wanna). 

This is tangential but finally, I think you could make self.move_number an `@property` of `len(self.move_history() + 1)` so you don't have to update it anymore. Obviously then take caution about where to append the next move to move_history.

trentzz replied:
> There's stuff common between the PDN making piece moves and the run loop with the bots that you could combine.

Yup sure

>  if self.verbose mode is on you also wanna print out the move and board display

verbose writes the games as a txt file, if it were to output to stdout it'll be huge

> self.move_number a property

Yeah sounds good 

>  take caution about where to append the next move to move_history.

This should be fine, I just add moves to the end of the move_history list

donren-leung replied:
> verbose writes the games as a txt file, if it were to output to stdout it'll be huge

I meant to the txt file not stdout...

> This should be fine, I just add moves to the end of the move_history list

I meant where to execute the appendment: at the start, end or somewhere in the middle -- because right now the appendment and increment of move counter are different places so you'll wanna be careful to not change any behaviour

donren-leung replied:
Moves are being made, just like in the run() loop, so I think it should get included in the txt file. It's just that they're mandated by the PDN so one of the lines could clarify, `Moved from x to x (Book Move)`

trentzz replied:
> so one of the lines could clarify, Moved from x to x (Book Move)

Okok

trentzz replied:
> I meant where to execute the appendment: at the start, end or somewhere in the middle

Yep I did fix this :D

trentzz commented:
Okok:
- Added the property for move_number
- Got rid of duplicated code for move_piece in Game
- Added check for complete game after import_pdn
- Added a test for properly raising RuntimeError when importing a complete game

donren-leung commented on checkers_bot_tournament/game.py:
```diff
@@ -78,21 +86,55 @@ def import_pdn(self, filename: str) -> None:
+            else:
+                if not self.board.is_valid_move(self.current_turn, move_obj):
+                    raise RuntimeError(
+                        f"Invalid move in import_pdn for colour: {str(self.current_turn)}, turn: {self.move_number + 1}, move: {move}"
```
What if there's a draw by repetition or anyhow a result in the middle of the PDN but it keeps going?

Ok I think the RuntimeError in line 103 should cover White/Black wins. Up to you if you wanna check for draws or leave it as-is cos it's an obscure case. In case you wanted to, you could yoink even more of the logic from make_move().

donren-leung commented:
Otherwise lgtm
